synapse/lib/stormtypes.py

```
@@ -2915,10 +2948,22 @@ async def toprim(valu, path=None):
         return valu
 
     if isinstance(valu, (tuple, list)):
-        return tuple([await toprim(v) for v in valu])
+        retn = []
+        for v in valu:
+            try:
+                retn.append(await toprim(v))
+            except s_exc.NoSuchType:
+                pass
+        return tuple(retn)
 
     if isinstance(valu, dict):
-        return {await toprim(k): await toprim(v) for (k, v) in valu.items()}
+        retn = {}
+        for k, v in valu.items():
+            try:
+                retn[k] = await toprim(v)
+            except s_exc.NoSuchType:
+                pass
```

silently dropping objects and changing the size of a dict feels scary.